specs/electra/validator.md

```diff
@@ -148,6 +185,24 @@ def prepare_execution_payload(state: BeaconState,
     )
 ```
 
+#### Execution Requests
+
+*[New in Electra]*
+
+1. The execution payload is obtained from the execution engine as defined above using `payload_id`. The response also includes a `execution_requests` entry containing a list of bytes. Each element on the list corresponds to one ssz list of requests as defined
+in [EIP-7685](https://eips.ethereum.org/EIPS/eip-7685). The index of each element in the array determines the type of request.
+2. Set `block.body.execution_requests = get_execution_requests(execution_requests)`, where:
+
+```python
+def get_execution_requests(execution_requests: List[bytes]) -> ExecutionRequests:
+    deposits = deserialize(List[DepositRequest, MAX_DEPOSIT_REQUESTS_PER_PAYLOAD], execution_requests[0])
+    withdrawals = deserialize(List[WithdrawalRequest, MAX_WITHDRAWAL_REQUESTS_PER_PAYLOAD], execution_requests[1])
+    consolidations = deserialize(List[ConsolidationRequest, MAX_CONSOLIDATION_REQUESTS_PER_PAYLOAD], 
+                                 execution_requests[2])
```

Comment:
I suppose the list of requests are not fixed size, it could be either 0 or 1 or 2 or 3, it's only guaranteed to sort ascending https://eips.ethereum.org/EIPS/eip-7685
so we need to based on the 1st byte to determine if it's DepositRequest, or WithdrawalRequest, or WithdrawalRequest instead of mapping `execution_requests[0]` for DepositRequest, `execution_requests[1]` for WithdrawalRequest  etc.

Reply:
No, the type is defined by the order in the array. There is no type on the first byte anymore.

Check https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md

Reply:
thanks for the clarification @lucassaldanha